gammapy/irf/effective_area_table.py

```diff
@@ -0,0 +1,572 @@
+
+        offset = offset.to('degree')
+        energy = energy.to('TeV')
+
+        # support energy=1Darray & offset=1Darray
+        if offset.shape != () and energy.shape != ():
+            val = np.zeros([len(offset), len(energy)])
+            for i in range(len(offset)):
+                val[i] = self._eval(offset[i], energy)
+        # default
+        else:
+            val = self._eval(offset=offset, energy=energy)
+
+        return Quantity(val, self.eff_area.unit)
+
+    def _eval(self, offset=None, energy=None):
+        method = self.interpolation_method
+        if(method == 'linear'):
+            val = self._linear(offset.value, np.log10(energy.value))
+        elif (method == 'spline'):
+            val = self._spline(offset.value, np.log10(energy.value)).squeeze()
+        else:
+            raise ValueError('Invalid interpolation method: {}'.format(method))
+        return Quantity(val, self.eff_area.unit)
+
+    def plot_energy_dependence(self, ax=None, offset=None, energy=None, **kwargs):
+        """Plot effective area versus energy for a given offset.
+        """
+        import matplotlib.pyplot as plt
+
+        ax = plt.gca() if ax is None else ax
+
+        if offset is None:
+            val = self.offset
+            offset = np.linspace(np.min(val), np.max(val), 5)
+
+        if energy is None:
+            energy = self.energy
+
+        for off in offset:
+            area = self.evaluate(off, energy)
+            label = 'offset = {:.1f}'.format(off)
+            plt.plot(energy, area.value, label=label, **kwargs)
+
+        plt.xlabel('Energy ({0})'.format(self.energy.unit))
+        plt.ylabel('Effective Area ({0})'.format(self.eff_area.unit))
+        plt.legend()
+
+        return ax
+
+    def plot_offset_dependence(self, ax=None, offset=None, energy=None, **kwargs):
+        """Plot effective area versus offset for a given energy
+        """
+        import matplotlib.pyplot as plt
+
+        ax = plt.gca() if ax is None else ax
+
+        if energy is None:
+            val = self.energy
+            energy = np.linspace(np.min(val), np.max(val), 4)
+
+        if offset is None:
+            offset = self.offset
+
+        for ener in energy:
+            area = self.evaluate(offset, ener)
+            label = 'energy = {:.1f}'.format(ener)
+            plt.plot(offset, area.value, label=label, **kwargs)
+
+        plt.xlabel('Offset ({0})'.format(self.offset.unit))
+        plt.ylabel('Effective Area ({0})'.format(self.eff_area.unit))
+        plt.legend()
+
+        return ax
+
+    def _prepare_linear_interpolator(self):
+        """Could be generalized for non-radial symmetric input files (N>2)
+        """
+
+        from scipy.interpolate import LinearNDInterpolator
+
+        x = self.offset.value
+        y = np.log10(self.energy.value)
+        coord = [(xx, yy) for xx in x for yy in y]
+
+        self._linear = LinearNDInterpolator(
```

Comment:
I'm not a big fan of these line breaks ... probably they were introduced by autopep8.
Can you use temp variables instead, i.e. in this case something like:
```python
vals = self.eff_area.value.flatten())
self._linear = LinearNDInterpolator(coord, vals)
```